spec.html

```diff
@@ -13799,6 +13892,74 @@ <h1>Runtime Semantics: ArgumentListEvaluation</h1>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-optional-chains">
+      <h1>Optional Chains</h1>
+      <emu-note>An optional chain is a chain of one or more property accesses and function calls, the first of which begins with the token `?.`.</emu-note>
+      <emu-clause id="sec-optional-chaining-evaluation">
+        <h1>Runtime Semantics: Evaluation</h1>
+        <emu-grammar>
+          OptionalExpression :
+            MemberExpression OptionalChain
+            CallExpression OptionalChain
+            OptionalExpression OptionalChain
+        </emu-grammar>
+        <emu-alg>
+          1. Let _baseExpression_ be the first child of this production (i.e., this |MemberExpression|, |CallExpression|, or |OptionalExpression|).
+          1. Let _baseReference_ be ? _baseExpression_.Evaluation().
+          1. Let _baseValue_ be ? GetValue(_baseReference_).
+          1. If _baseValue_ is *undefined* or *null*, then
+            1. Return *undefined*.
+          1. Let _optionalChain_ be this |OptionalChain|.
+          1. Return ? _optionalChain_.ChainEvaluation(_baseValue_, _baseReference_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-optional-chaining-chain-evaluation">
+        <h1>Runtime Semantics: ChainEvaluation</h1>
+        <p>With parameters _baseValue_ and _baseReference_.</p>
+        <emu-grammar>OptionalChain : `?.` `[` Expression `]`</emu-grammar>
+        <emu-alg>
+          1. If the code matched by this production is strict mode code, let _strict_ be *true*, else let _strict_ be *false*.
+          1. Return ? EvaluateDynamicPropertyAccess(_baseValue_, |Expression|, _strict_).
+        </emu-alg>
+        <emu-grammar>OptionalChain : `?.` IdentifierName</emu-grammar>
+        <emu-alg>
+          1. If the code matched by this production is strict mode code, let _strict_ be *true*, else let _strict_ be *false*.
+          1. Return ? EvaluateStaticPropertyAccess(_baseValue_, |IdentifierName|, _strict_).
+        </emu-alg>
+        <emu-grammar>OptionalChain : `?.` Arguments</emu-grammar>
+        <emu-alg>
+          1. Let _thisChain_ be this production.
+          1. Let _tailCall_ be IsInTailPosition(_thisChain_).
+          1. Return ? EvaluateCall(_baseValue_, _baseReference_, |Arguments|, _tailCall_).
+        </emu-alg>
+        <emu-grammar>OptionalChain : OptionalChain `[` Expression `]`</emu-grammar>
+        <emu-alg>
+          1. Let _optionalChain_ be this |OptionalChain|.
```

Comment:
```suggestion
          1. Let _optionalChain_ be |OptionalChain|.
```

The `this |Nonterminal|` wording refers to the the entire production, rather than one of the nonterminals on the RHS. Saying `Let _optionalChain_ be |OptionalChain|` would refer to the one on the RHS. See for example [the Evaluation semantics for](https://tc39.es/ecma262/#sec-function-calls-runtime-semantics-evaluation) `CallExpression : CallExpression Arguments`, which has exactly the same ambiguity, and which in step 1 uses a bare `CallExpression` to mean the one on the RHS and in step 3 uses `this CallExpression` to mean the full production.

Here this line only makes sense if it is intended to refer to the one one the RHS rather than the full production.

(Ditto most other uses of `this |NonTerminal|` in this PR.)

Reply:
This was fixed in https://github.com/tc39/proposal-optional-chaining/pull/120 but must have not been propagated to this PR.